storage/memoized_iterator.go

```diff
@@ -108,15 +103,14 @@ func (b *MemoizedSeriesIterator) Next() chunkenc.ValueType {
 		return chunkenc.ValNone
 	case chunkenc.ValFloat:
 		b.prevTime, b.prevValue = b.it.At()
-		b.prevHistogram = nil
 		b.prevFloatHistogram = nil
 	case chunkenc.ValHistogram:
 		b.prevValue = 0
-		b.prevTime, b.prevHistogram = b.it.AtHistogram()
-		_, b.prevFloatHistogram = b.it.AtFloatHistogram()
+		ts, h := b.it.AtHistogram()
```

Comment:
If we are removing support for `AtHistogram`, can we call `b.it.AtFloatHistogram` directly to avoid allocating data twice? We will never need the original histogram and we can decode to a float histogram directly.

Reply:
I wasn't sure if we could depend on the underlying iterator being able to perform the float histogram conversion directly but looking closer that does seem to be the case--I'll make the change!

Reply:
Made the change as well as adding a comment clarifying the assumption.